src/App.js

```diff
@@ -0,0 +1,25 @@
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { Outlet } from 'react-router-dom';
+import { Navigation } from './features';
+import fetchMission from './redux/missions/missionThunk';
+import fetchRocket from './redux/rockets/rocketThunk';
+import './App.css';
+
+function App() {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchMission());
+    dispatch(fetchRocket());
+  }, [dispatch]);
```

Comment:
- [ ] You did a great job for the most part. :+1: Let's work on some more details to achieve the project requirements. We are supposed to make API calls from their corresponding components, which means `fetchMission` should be dispatched from the Missions component while `fetchRocket` should be dispatched from the Rockets component. Kindly remove the `useEffect` from the App component. 🥸️
While doing so, please also make sure that you are not making calls to the API every time the user navigates between the pages. You can achieve this by checking if the current piece of state is empty or not. 🙂️

**Project requirements**
![Screenshot from 2023-04-13 11-54-39](https://user-images.githubusercontent.com/17243325/231711636-b2c298f5-cc0a-4b3c-a432-9076f408ba6d.png)